subsys/testsuite/ztest/src/ztest_new.c

```diff
@@ -64,8 +64,13 @@ static ZTEST_BMEM enum ztest_status test_status = ZTEST_STATUS_OK;
 
 extern ZTEST_DMEM const struct ztest_arch_api ztest_api;
 
+static void __ztest_show_suite_summary(void);
+
 void end_report(void)
 {
+	if (IS_ENABLED(CONFIG_ZTEST_SUMMARY)) {
```

Comment:
Instead of putting a condition on this call, could you put a condition inside `__ztest_show_suite_summary`?

Relatedly, it looks like there is an opportunity to simplify the config check inside the summary functions. `__ztest_show_suite_summary` conditionally calls `__ztest_show_suite_summary_verbose` or `__ztest_show_suite_summary_oneline`. Then, `__ztest_show_suite_summary_verbose` calls `__ztest_show_suite_summary_oneline` as its first action.

I think this would be simpler to read as one early-return config check at the beginning of `__ztest_show_suite_summary` to decide whether to print a summary at all and another early-return config check near the beginning of `__zteset_show_suite_summary_verbose` to decide whether to do anything besides the one-line summary.

Reply:
done